### PR TITLE
More efficient  removal

### PR DESCRIPTION
**Pull request by zhu733756**
Signed-off-by: zhu733756 <talonzhu@yunify.com>

Currently, the removal script takes a lot of time.

### CODE REVIEW

**Comment by zhu733756** [comment]
cc

File: hack/delete-tekon-and-knative.sh [hunk 1]
```diff
@@ -1,133 +1,27 @@
...
-kubectl delete validatingwebhookconfiguration.admissionregistration.k8s.io validation.webhook.eventing.knative.dev
-kubectl delete mutatingwebhookconfiguration.admissionregistration.k8s.io sinkbindings.webhook.sources.knative.dev
\ No newline at end of file
+version=v0.23.0
+if [ $1 ]; then
+    version=$1
+fi
+
+# delete Tekton pipeline
+kubectl delete --filename https://openfunction.sh1a.qingstor.com/tekton/pipeline/${version}/release.yaml
```
Comment (benjaminhuo) [inline hack/delete-tekon-and-knative.sh:9]: If you take a look at https://github.com/OpenFunction/OpenFunction/blob/main/hack/deploy-tekon-and-knative.sh#L39
You'll find out that we're trying to use google storage for non-China deployment and use qingstor for deployment in China, I think the original method is fine.

**Review by benjaminhuo — COMMENTED** [review]
See my comments above

File: hack/delete-tekon-and-knative.sh (hunk 1 above)
Comment (zhu733756) [reply hack/delete-tekon-and-knative.sh:9]: The original method blocked at `kubectl delete ns xxx`  for minutes of time. I think the removal processing can be improved by using google storage files. The Chinese user blocked by pulling images. The removal processing just use the defination of the yamls. I will convert it.

**Comment by zhu733756** [comment]
By the way, take a look at https://github.com/OpenFunction/OpenFunction/blob/main/hack/deploy-tekon-and-knative.sh, the chinese users dont apply eventing-crds, however,  the google storage users use this yaml. Is this  a mistake?
```
# Install the required custom resource definitions (CRDs)
kubectl apply -f https://github.com/knative/eventing/releases/download/v0.23.0/eventing-crds.yaml
```
```
#kubectl apply -f https://openfunction.sh1a.qingstor.com/knative/eventing/v0.23.0/eventing-crds.yaml
```

**Comment by tpiperatgod** [comment]
> By the way, take a look at https://github.com/OpenFunction/OpenFunction/blob/main/hack/deploy-tekon-and-knative.sh, the chinese users dont apply eventing-crds, however, the google storage users use this yaml. Is this a mistake?
> 
> ```
> # Install the required custom resource definitions (CRDs)
> kubectl apply -f https://github.com/knative/eventing/releases/download/v0.23.0/eventing-crds.yaml
> ```
> 
> ```
> #kubectl apply -f https://openfunction.sh1a.qingstor.com/knative/eventing/v0.23.0/eventing-crds.yaml
> ```

*-core.yaml already contains the definition of crd resources, so we can actually ignore the *-crds.yaml installation for eventing, serving

And here is indeed a mistake, should we comment them all?

File: hack/delete-tekon-and-knative.sh (hunk 1 above)
Comment (benjaminhuo) [reply hack/delete-tekon-and-knative.sh]: @wanjunlei Pls take a look at your original script to see how to fix the delete ns issue.

Comment (wanjunlei) [reply hack/delete-tekon-and-knative.sh]: The reason for blocking at kubectl delete ns xxx is It need delete all resources i the ns before delete ns. The solution is to delete all resources in the ns, and then delete ns.

**Comment by benjaminhuo** [comment]
> > By the way, take a look at https://github.com/OpenFunction/OpenFunction/blob/main/hack/deploy-tekon-and-knative.sh, the chinese users dont apply eventing-crds, however, the google storage users use this yaml. Is this a mistake?
> > ```
> > # Install the required custom resource definitions (CRDs)
> > kubectl apply -f https://github.com/knative/eventing/releases/download/v0.23.0/eventing-crds.yaml
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > #kubectl apply -f https://openfunction.sh1a.qingstor.com/knative/eventing/v0.23.0/eventing-crds.yaml
> > ```
> 
> *-core.yaml already contains the definition of crd resources, so we can actually ignore the *-crds.yaml installation for eventing, serving
> 
> And here is indeed a mistake, should we comment them all?

yes pls

**Comment by wanjunlei** [comment]
> > > By the way, take a look at https://github.com/OpenFunction/OpenFunction/blob/main/hack/deploy-tekon-and-knative.sh, the chinese users dont apply eventing-crds, however, the google storage users use this yaml. Is this a mistake?
> > > ```
> > > # Install the required custom resource definitions (CRDs)
> > > kubectl apply -f https://github.com/knative/eventing/releases/download/v0.23.0/eventing-crds.yaml
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > ```
> > > #kubectl apply -f https://openfunction.sh1a.qingstor.com/knative/eventing/v0.23.0/eventing-crds.yaml
> > > ```
> > 
> > 
> > *-core.yaml already contains the definition of crd resources, so we can actually ignore the *-crds.yaml installation for eventing, serving
> > And here is indeed a mistake, should we comment them all?
> 
> yes pls

There will have an error that crd `images.caching.internal.knative.dev` not found when deploying serving-core.yaml,  because the time between creating cr and creating crd is too short, apiserver has no time to synchronize data. 
Therefore I kept the deployment of serving-crds.yaml.

**Pull request closed without merging**